core/src/test/scala/org/apache/spark/h2o/converters/DataFrameConverterTest.scala

```diff
@@ -490,21 +490,21 @@ class DataFrameConverterTest extends FunSuite with SharedH2OTestContext {
   }
 
   test("Expand schema with array") {
-    import spark.implicits._
     val num = 5
-    val values = (1 to num).map(x => PrimitiveB(1 to x))
-    val df = sc.parallelize(values).toDF
+    val rdd = sc.parallelize(1 to num).map(x => Row(1 to x))
+    val schema = StructType(StructField("f", ArrayType(IntegerType, containsNull = false), nullable = false) :: Nil)
+    val df = spark.createDataFrame(rdd, schema)
 
     val (flattenDF, maxElementSizes, expandedSchema) = getSchemaInfo(df)
 
     val metadatas = expandedSchema.map(f => f.metadata)
 
-    assert(expandedSchema === Vector(
-      (StructField("f0", IntegerType, nullable = false, metadatas.head)),
-      (StructField("f1", IntegerType, nullable = false, metadatas(1))),
-      (StructField("f2", IntegerType, nullable = false, metadatas(2))),
-      (StructField("f3", IntegerType, nullable = false, metadatas(3))),
-      (StructField("f4", IntegerType, nullable = false, metadatas(4)))))
+    assert(expandedSchema === Array(
+      (StructField("f_0", IntegerType, nullable = false, metadatas.head)),
```

Comment:
I'm a bit slower. Please can you elaborate more on this comment? Is this comment related to this test?

Reply:
Sorry, my explanation was bad. I just wanted to make sure that changing the separator does not influence the user code as we will be returning the original frame with attached predictions

Reply:
It will be a breaking change for users using structured dataframes. The original code was returning a flattened dataset with prediction. With this change, SW will return an original structured data frame with predictions. But `_` should be just an internal thing, if a user use just pipeline API. If he/she plays with conversions between data frames and H2O frames, the he/she can experience  some changes.

Reply:
Yup, this is fine :)

Reply:
There might be also problems with MOJOs that were created with previous SW versions. It seems that column renaming will change prediction behavior of old MOJOs in the new version.

Reply:
Good point -> I'm thinking of a good approach. The backwards compatible solution is to keep the separator as `.` as that would ensure the behaviour is the same as before.

Reply:
ok, I will change it globally to `.`

Reply:
One more comment -> I actually think `_` is better choice as that dot usually implies the nested structures, but to introduce `_` we should deprecate the old behaviour first.

Reply:
Let's introduce `_` as a default delimiter of the separate flattening estimator.

Reply:
Nice solution, the current flattening will be removed anyway after introduction of the flattening estimator.